### PR TITLE
Add support for EIP1559

### PR DESCRIPTION
**Pull request by evgenyboxer**
# Description

Adds support for EIP1559 type transactions.

Here is an example of a tx I made with the change - https://etherscan.io/tx/0xf2d62b204da7f2c0500fb0844821c627ca177a3d8c9cb15e1402be2041c38dd3

Fixes # (issue)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Other

## How should this be tested?

Please provide instructions so we can test. Please also list any relevant details for your test configuration.

- [ ] Make a trade

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code where relevant, particularly in hard-to-understand areas
- [x] My changes generate no new console warnings
- [x] The base of this PR is `master` if hotfix, `develop` if not

### CODE REVIEW

File: src/lib/utils/balancer/web3.ts [hunk 1]
```diff
@@ -42,10 +42,23 @@ export async function sendTransaction(
     const gasLimit = gasLimitNumber.toNumber();
     overrides.gasLimit = Math.floor(gasLimit * (1 + GAS_LIMIT_BUFFER));
 
-    if (USE_BLOCKNATIVE_GAS_PLATFORM && overrides.gasPrice == null) {
+    if (
+      USE_BLOCKNATIVE_GAS_PLATFORM &&
+      overrides.gasPrice == null &&
+      overrides.maxFeePerGas == null &&
+      overrides.maxPriorityFeePerGas == null
```
Comment (evgenyboxer) [inline src/lib/utils/balancer/web3.ts:49]: if any of these are set I think the intent is for manual control over the gas price.

Comment (1saf) [reply src/lib/utils/balancer/web3.ts:49]: If any are changed then shouldn't we use an OR condition

File: src/services/gas-price/providers/blocknative.provider.ts [hunk 2]
```diff
@@ -64,7 +67,13 @@ export default class BlocknativeProvider {
       }
 
       // gas price is in gwei
-      return gasPrice != null ? gasPrice.price * GWEI_UNIT : null;
+      return gasPrice != null
```
Comment (1saf) [inline src/services/gas-price/providers/blocknative.provider.ts:70]: nit: i'd prefer a traditional if statement here compared to a ternary for the purpose of readability

**Review by 1saf — APPROVED** [review]

File: src/lib/utils/balancer/web3.ts (hunk 1 above)
Comment (evgenyboxer) [reply src/lib/utils/balancer/web3.ts:49]: I mean, I need all of them to be `null` to proceed (or more likely `undefined`) - but that will match both anyway.

File: src/services/gas-price/providers/blocknative.provider.ts (hunk 2 above)
Comment (evgenyboxer) [reply src/services/gas-price/providers/blocknative.provider.ts]: yeah I see what you mean, its a bit hard to read in this instance - I've pushed a fix for it

**Pull request merged**